CMakeLists.txt

```diff
@@ -1811,12 +1811,22 @@ if(CONFIG_CHECK_INIT_PRIORITIES)
   list(APPEND
     post_build_commands
     COMMAND ${PYTHON_EXECUTABLE} ${ZEPHYR_BASE}/scripts/build/check_init_priorities.py
-      --build-dir ${PROJECT_BINARY_DIR}/..
-      --edt-pickle ${EDT_PICKLE}
+      --elf-file=${ZEPHYR_BINARY_DIR}/${KERNEL_ELF_NAME}
       ${fail_on_warning}
     )
 endif()
 
+if(NOT CMAKE_C_COMPILER_ID STREQUAL "ARMClang")
```

Comment:
not fond, of it, but I see your reasoning here.
https://github.com/zephyrproject-rtos/zephyr/pull/62459#discussion_r1331431765

and users could want to run the target even with the kconfig to `n`.
I don't think this specific feature is strong enough to have something along `TOOLCHAIN_CREATES_GOOD_ELF` or similar.

So accepted for now, and let's see if we can find a way with armclang to support the check init feature.

Reply:
Yeah, I'm sure it can be fixed, those stripping options in `cmake/bintools/armclang/elfconvert_command.cmake` looks suspicious to me. It'd be cool to figure a way of having a CI run with these compilers so we don't depend on individuals to test stuff though.